### PR TITLE
Revise documentation of bad UID remapping error

### LINKED ISSUE #1639: Docs missing valid UID range
This issue is about <https://circleci.com/docs/2.0/high-uid-error/> (source file: <https://github.com/circleci/circleci-docs/blob/master/jekyll/_cci2/high-uid-error.md>)

This page does a good job of explaining why high user IDs are problematic. However, it simply says that you must "change the ownership of the file" - without indicating what a _valid uid / gid range actually is_. It'd be good to include that so image maintainers don't have to guess.

(as an aside, just running `cat /etc/passwd` may be enough in many cases to find the high ID).

### LINKED ISSUE #1786: Bad assumption
This issue is about <https://circleci.com/docs/2.0/high-uid-error/> (source file: <https://github.com/circleci/circleci-docs/blob/master/jekyll/_cci2/high-uid-error.md>)

I hit the error yesterday and whilst the page was very helpful in explaining the problem, it didn't get me to where I could fix it.

Neither the find command suggested, nor any other approach, could find me any file with a too-high UID.

In the end, I discovered that it was one of the interim steps in building the container that was creating the bad file, but that file was gone by the time the final image was generated.

To fix it, I had to modify the "RUN" step in the Dockerfile, adding "&& chown -R root:root /root" to the problem step. This meant that the problem file was created and fixed all in one step and thus didn't create a bad interim.

The assumption in the docs is that the bad file will still exist in the image when it's run, but this is clearly not necessarily the case. Updating to explain that a bad file can be created, removed, and still cause a problem; and how to debug & fix this, would be very helpful.

### PR DESCRIPTION
**Pull request by smart-alek**
Full revision of the infamous high UUID error. Fixes #1786 and fixes #1639

### CODE REVIEW

File: jekyll/_cci2/high-uid-error.md [hunk 1]
```diff
@@ -8,54 +8,70 @@ order: 21
...
+In the example presented at the top of this document,
+this value is `1000000`.
+Next, start the container and look for the files which receive that invalid value.
+
+One way to do this is to use the `find` command,
+as shown below:
+
+```bash
 # Start a shell inside the container
 $ docker run -it circleci/doc-highid sh
```
Comment (kimh) [inline jekyll/_cci2/high-uid-error.md]: I think `circleci/doc-highid` doesn't make sense if you delete a link to `https://hub.docker.com/r/circleci/doc-highid`.

File: jekyll/_cci2/high-uid-error.md [hunk 2]
```diff
@@ -68,4 +84,14 @@ $ ls -ln file-with-high-id
...
+After locating the offending file,
+change its ownership
+and re-create the image.
+
+**Note:**
+It is possible for an invalid file to be generated and removed
+while a container is building.
+In this case,
+you may have to modify the `RUN` step in the Dockerfile itself.
+Adding `&& chown -R root:root /root` to the problem step
```
Comment (kimh) [inline jekyll/_cci2/high-uid-error.md]: Is it reasonable to assume that users can identify which step to add this command?

File: jekyll/_cci2/high-uid-error.md (hunk 1 above)
Comment (kimh) [inline jekyll/_cci2/high-uid-error.md]: Maybe we should address https://github.com/circleci/circleci-docs/issues/1639 as well in this PR?

Comment (smart-alek) [reply jekyll/_cci2/high-uid-error.md]: Whoops — yeah, I'll re-add that link.

File: jekyll/_cci2/high-uid-error.md (hunk 2 above)
Comment (smart-alek) [reply jekyll/_cci2/high-uid-error.md]: This gets into some philosophical territory for me, actually: how deep do we want to go to explain things? Is it _our fault_ that this bug even exists? If it is, then I'd say we need to provide a complete workaround.

**Pull request merged**